sonic-ycabled/ycable/ycable_utilities/y_cable_helper.py

```diff
@@ -458,34 +458,48 @@ def connect_channel(channel, stub, port):
         else:
             break
 
-def create_channel(type_chan, level, kvp, soc_ip, port, asic_index, fwd_state_response_tbl, is_async):
+def wait_for_state_change(channel_connectivity, port):
+    # Initialize YcableChannelStateTableHelper only once
+    if not hasattr(wait_for_state_change, 'table_helper'):
+        wait_for_state_change.table_helper = {}
 
-    # Helper callback to get an channel connectivity state
-    def wait_for_state_change(channel_connectivity):
-        if channel_connectivity == grpc.ChannelConnectivity.TRANSIENT_FAILURE:
-            helper_logger.log_notice("gRPC port {} state changed to TRANSIENT_FAILURE".format(port))
-            # for connectivity state to FAILURE/IDLE report a failure
-            fvs_updated = swsscommon.FieldValuePairs([('response', 'failure')])
-            fwd_state_response_tbl[asic_index].set(port, fvs_updated)
-            grpc_port_connectivity[port] = "TRANSIENT_FAILURE"
-
-        if channel_connectivity == grpc.ChannelConnectivity.CONNECTING:
-            helper_logger.log_notice("gRPC port {} state changed to CONNECTING".format(port))
-            grpc_port_connectivity[port] = "CONNECTING"
-        if channel_connectivity == grpc.ChannelConnectivity.READY:
-            helper_logger.log_notice("gRPC port {} state changed to READY".format(port))
-            grpc_port_connectivity[port] = "READY"
-        if channel_connectivity == grpc.ChannelConnectivity.IDLE:
-            helper_logger.log_notice("gRPC port {} state changed to IDLE".format(port))
-            # for connectivity state to FAILURE/IDLE report a failure
-            fvs_updated = swsscommon.FieldValuePairs([('response', 'failure')])
-            fwd_state_response_tbl[asic_index].set(port, fvs_updated) 
-            grpc_port_connectivity[port] = "IDLE"
-
-        if channel_connectivity == grpc.ChannelConnectivity.SHUTDOWN:
-            helper_logger.log_notice("gRPC port {} state changed to SHUTDOWN".format(port))
-            grpc_port_connectivity[port] = "SHUTDOWN"
+    if wait_for_state_change.table_helper.get(port, None) is None:
+        wait_for_state_change.table_helper[port] = y_cable_table_helper.YcableChannelStateTableHelper()
```

Comment:
@vdahiya12 why each port need to have handler for this table?

Reply:
its a more cleaner way to handle this if we have a per port based swsscommon handle. 
for each port we have a seperate channel, stub in ycabled today. 

As channel.subscribe launches into a new separate thread for every single port, its better to not share this across multiple ports, or else we run into same problem